### PR TITLE
Input/sqs/arbitrary json msg

### PR DESCRIPTION
**Pull request by arl**
#### :question: What

Add support to extract a S3 path from any notification, as long as it's in JSON.
Supported format:
 - plain
 - sns
 - json (new format)
 - s3::ObjectCreated (new format)

When `json` is used, a new `MessageExpression` field must contain a valid jmespath expression.
This can be extended in the future if we want to support, say, xml messages or anything else. 

Also, the `sns` format is kept for backward-compatbility, even though it could be rewritten as:
```toml
MessageFormat = "json"
MessageExpression = "Message" # takes the path from the Message field of the notification payload
```
Internally, `sns` is rewritten as above.

Also, add the `s3::ObjectCreated` predfined format since it's something that is in use and the jmespath expression is not trivial

This also removes the `sqs.lag` metric. It was not documented, not used (in any NextRoll dashboard) but moreover not all messages contains a timestamp, and the timestamp may mean different things anyway, or be in any format/timezone.
I think there are more specific metrics, such as the age of oldest message for an SQS queue, that can be used to the same effect (i.e knowing the delay/lag between producer and consumer).
We could re-add this feature once we have sorted the problems above out (timestamp meaning, format, etc.) if necessary.

#### :hammer: How to test

1. List all steps necessary;
2. To test this pull request.

#### :white_check_mark: Checklists

_This section contains a list of checklists for common uses, please delete the checklists that are useless for your current use case (or add another checklist if your use case isn't covered yet)._

- [x] Is there unit/integration test coverage for all new and/or changed functionality added in this PR?
- [x] Have the changes in this PR been functionally tested?
- [] Have new components been added to the related `all.go` files?
- [] Have new components been added to the documentation website?
- [x] Has `make gofmt-write` been run on the code?
- [x] Has `make govet` been run on the code? Has the code been fixed accordingly to the output?
- [x] Have the changes been added to the [CHANGELOG.md](./CHANGELOG.md) file?
- [x] Have the steps in [CONTRIBUTING.md](./CONTRIBUTING.md) been followed to update a Go module?

### CODE REVIEW

**Review by lucarin91 — APPROVED** [review]
Very useful change :tada:. LGTM, I added only some minor comments.

File: input/sqs.go [hunk 1]
```diff
@@ -3,93 +3,177 @@ package input
...
 
 var SQSDesc = baker.InputDesc{
 	Name:   "SQS",
 	New:    NewSQS,
 	Config: &SQSConfig{},
-	Help: "This input listens on multiple SQS queues for new incoming log files\n" +
-		"on S3; it is meant to be used with SQS queues popoulated by SNS.\n" +
-		"It never exits.\n",
+	Help: `This input listens on multiple SQS queues for new incoming log files
+on S3; it is meant to be used with SQS queues populated by SNS.
```
[the remark this answers is not in the document]
Comment (arl) [reply input/sqs.go]: done

**Review by lucarin91 — COMMENTED** [review]
I think there is a typo in the doc.

File: input/sqs.go [hunk 2]
```diff
@@ -3,93 +3,177 @@ package input
...
-		"It never exits.\n",
+	Help: `This input listens on multiple SQS queues for new incoming files on S3.
+It can be used with SQS queues subscribed to SNS topics (raw_message_delivery subscription).
+It never exits.
+
+Supported formats (MessageFormat):
+ - "plain": the message payload is a plain S3 path.
+ - "sns": the payload is an SNS notification
+ - "s3::ObjectCreated": the payload is the message generated by an S3::ObjectCreated notification
+ - "json": the message is an arbitrary JSON string, the S3 path is extracted from it via a [jmespath](https://jmespath.org/) expression specified in MessageExpression. The SQS subscription
```
[the remark this answers is not in the document]
Comment (arl) [reply input/sqs.go:36]: thanks

**Review by lucarin91 — APPROVED** [review]
:+1:

**Pull request merged**